atlp-rwanda/atlp-pulse-mobile
# ft(#4): Reset password

### What does this PR do?

This pull request implements a new feature that allows users to reset their passwords by sending a password reset email. When a user clicks the reset password link in the email, the app will open and redirect them to the setNewPassword screen.

### Description of Task to be completed?

- Create a new password reset endpoint: Implement a new endpoint that handles password reset requests. This endpoint should generate a unique password reset token and send an email to the user containing a link to a password reset page.
- Implement password reset functionality: Create a new screen or page where users can enter their new password. When a user submits the new password, the app should verify the password reset token and update the user's password if the token is valid.
- Handle email delivery: Integrate an email service (e.g., SendGrid, Mailgun) to send password reset emails. Ensure that the email template includes a clear link to the password reset page.
- Implement security measures: Implement security measures to protect password reset tokens and prevent unauthorized password resets. Consider using techniques like token expiration, token invalidation, and rate limiting.

### How should this be manually tested?

1. Create a new user account.
2. Request a password reset.
3. Click the password reset link in the email.
4. Enter a new password on the setNewPassword screen.
5. Verify that the password reset is successful by logging in with the new password.
6. Test the password reset functionality with invalid email addresses, expired tokens, and rate limiting.

### Screenshots:
<div style="display: flex; flex-wrap: wrap;">
    <img src="https://github.com/user-attachments/assets/a35d5b85-3505-4297-ba19-c63c6b1946c1" alt="Screenshot 1" height="480px">
        <img src="https://github.com/user-attachments/assets/677e7503-d04e-4182-8014-c12a9c98bab9" alt="Screenshot 1" height="480px">
    <img src="https://github.com/user-attachments/assets/f14071fb-f2c2-459a-85de-0b3a99fa363d" alt="Screenshot 2" height="480px">
        <img src="https://github.com/user-attachments/assets/2aa6fcc0-4772-4dd7-9f00-97f6150f1d9c" alt="Screenshot 1" height="480px">
</div>
